### PR TITLE
Add Support for FCM

### PR DESCRIPTION
**Pull request by alexandreferreira**
I saw the last try to add FCM support and was repeating too much code, so a resolve to just adapt the GCM code and add a new column with cloud type, so when the messages is send, the code decides which cloud message is to call.

Thanks.

### CODE REVIEW

**Comment by andhieka** [comment]
Looks like a comprehensive pull request. Have you tested this on your own server? @alexandreferreira

**Comment by jleclanche** [comment]
@jamaalscarlett I'm giving you ownership of this PR. I'm currently traveling and unable to look into it very much.

@alexandreferreira thanks for the work - make sure you follow the guidelines in [CONTRIBUTING.md](https://github.com/jleclanche/django-push-notifications/blob/master/CONTRIBUTING.md) in order to get it merged sooner.

**Comment by jamaalscarlett** [comment]
@alexandreferreira Thanks for the PR!  Before I start commenting, could you please squash these 3 commits into 1.  Thanks

**Comment by alexandreferreira** [comment]
@jamaalscarlett Ok, I think a squashed right.

@andhieka I'm using this version in my project.

@jleclanche I check the guide lines, I think thats ok.

File: push_notifications/gcm.py [hunk 1]
```diff
@@ -201,7 +245,38 @@ def gcm_send_bulk_message(registration_ids, data, **kwargs):
...
 			for chunk in _chunks(registration_ids, max_recipients):
-				ret.append(_gcm_send_json(chunk, data, **kwargs))
+				ret.append(_cm_send_json(chunk, data, cloud_type="GCM", **kwargs))
 			return ret
 
-	return _gcm_send_json(registration_ids, data, **kwargs)
+	return _cm_send_json(registration_ids, data, cloud_type="GCM", **kwargs)
+
+
+def fcm_send_bulk_message(registration_ids, data, **kwargs):
```
Comment (alexandreferreira) [inline push_notifications/gcm.py]: There's no need, I separated to be explicit that configs e prepares for GCM and FCM are different.
If separated in different methods any custom properties or behaviors in FCM will be easier to do

Comment (alexandreferreira) [inline push_notifications/gcm.py]: OK, I'll add a if/else

**Comment by alexandreferreira** [comment]
@jamaalscarlett I changed the logic to work with if/else, it's simpler now, thanks for the tip :)

**Comment by jleclanche** [comment]
@alexandreferreira Thanks for your PR. I have merged it, with a bunch of code style fixes and reduced the max_length of the cloud message type to 3 as it's unlikely to ever be anything else (as a sidenote, that should have been a PositiveSmallIntegerField imho, but it's not important enough at this point to matter).

I have some concerns with all the code duplication especially in the tests. But I'll leave this up to @jamaalscarlett :)

**Pull request closed without merging**